### PR TITLE
Refactoring the commands

### PR DESCRIPTION
**Pull request by SweeXordious**
- fixes the generate command
- hides unnecessary options from the new command
- Refactors the commands
- Adds `epirus openapi new/import/generate/import`

### CODE REVIEW

**Comment by SweeXordious** [comment]
@xaviarias Can you please review this ?

File: src/main/java/io/epirus/console/project/NewProjectCommand.java [hunk 1]
```diff
@@ -127,16 +118,21 @@ public void run() {
...
+                                                projectName,
+                                                packageName,
+                                                outputDir,
+                                                Collections.emptyList(),
+                                                Collections.emptyList(),
+                                                addressLength,
+                                                contextPath != null
+                                                        ? StringUtils.removeEnd(contextPath, "/")
+                                                        : projectName,
+                                                true,
```
Comment (xaviarias) [inline src/main/java/io/epirus/console/project/NewProjectCommand.java:131]: This smells a bit weird... two places in the code where the values are always `true` ? 🤔

**Review by xaviarias — APPROVED** [review]

File: src/main/java/io/epirus/console/project/NewProjectCommand.java (hunk 1 above)
Comment (SweeXordious) [reply src/main/java/io/epirus/console/project/NewProjectCommand.java:131]: This will change after having the `web3j-openapi-gradle-plugin` working

Comment (SweeXordious) [reply src/main/java/io/epirus/console/project/NewProjectCommand.java:131]: Won't merge this until we release a new version of web3j-openapi

**Review by AlexandrouR — APPROVED** [review]

File: src/main/java/io/epirus/console/project/BaseProjectOptions.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * specific language governing permissions and limitations under the License.
+ */
+package io.epirus.console.project;
+
+import picocli.CommandLine;
+
+import static picocli.CommandLine.Help.Visibility.ALWAYS;
+
+public class BaseProjectOptions {
+    @CommandLine.Option(
```
Comment (xaviarias) [inline src/main/java/io/epirus/console/project/BaseProjectOptions.java:20]: Please import `CommandLine.Option` when using multiple times.

File: src/main/java/io/epirus/console/project/ImportProjectCommand.java [hunk 3]
```diff
@@ -43,60 +39,13 @@
         footer = "Epirus CLI is licensed under the Apache License 2.0")
 public class ImportProjectCommand implements Runnable {
 
-    @CommandLine.ArgGroup() ProjectType projectType = new ProjectType();
```
Comment (xaviarias) [inline src/main/java/io/epirus/console/project/ImportProjectCommand.java:46]: Please import `CommandLine.Option`and `CommandLine.ArgGroup`.

File: src/main/java/io/epirus/console/project/NewProjectCommand.java [hunk 4]
```diff
@@ -48,59 +37,10 @@
         optionListHeading = "%nOptions:%n",
         footerHeading = "%n",
         footer = "Epirus CLI is licensed under the Apache License 2.0")
-public class NewProjectCommand implements Runnable {
+public class NewProjectCommand extends ProjectOptions implements Runnable {
 
-    @CommandLine.ArgGroup() ProjectType projectType = new ProjectType();
```
Comment (xaviarias) [inline src/main/java/io/epirus/console/project/NewProjectCommand.java:53]: Please import `CommandLine.Option`, `CommandLine. Parameters ` and `CommandLine.ArgGroup`.

File: src/main/java/io/epirus/console/project/ProjectOptions.java [hunk 5]
```diff
@@ -14,22 +14,24 @@
 
 import picocli.CommandLine;
 
-public class ProjectType {
+import static picocli.CommandLine.Help.Visibility.ALWAYS;
+
+public class ProjectOptions extends BaseProjectOptions {
     @CommandLine.Option(
```
Comment (xaviarias) [inline src/main/java/io/epirus/console/project/ProjectOptions.java:20]: Import `CommandLine.Option`.

File: src/main/java/io/epirus/console/wrapper/subcommand/SolidityGenerateCommand.java [hunk 6]
```diff
@@ -85,7 +85,7 @@
 
     @CommandLine.Option(
```
Comment (xaviarias) [inline src/main/java/io/epirus/console/wrapper/subcommand/SolidityGenerateCommand.java:86]: Import `CommandLine.Option`

File: src/main/resources/project/build.gradleOpenAPI.template [hunk 7]
```diff
@@ -0,0 +1,45 @@
+buildscript {
+    repositories {
+        mavenCentral()
+        maven { url 'https://repo.gradle.org/gradle/libs-releases' }
+        maven { url 'https://dl.bintray.com/hotkeytlt/maven' }
+        maven { url "https://plugins.gradle.org/m2/" }
+        mavenLocal()
+    }
+    dependencies {
+        classpath 'org.web3j.openapi:web3j-openapi-gradle-plugin:4.5.12.4'
```
Comment (xaviarias) [inline src/main/resources/project/build.gradleOpenAPI.template:10]: Let's use 4.6.5 for latest release :)

**Review by xaviarias — CHANGES_REQUESTED** [review]

**Pull request merged**